ui/app/components/clients/current.js

```diff
@@ -84,6 +84,7 @@ export default class Current extends Component {
     // value comes in as [namespace0]
     this.selectedNamespace = value;
     if (!value) {
+      this.authMethodOptions = [];
       // on clear, also make sure auth method is cleared
       this.selectedAuthMethod = null;
```

Comment:
This line should remove the auth method filter when namespace is deselected, so not sure why that wasn't working 🤔

Reply:
Yes that's what I added to fix the issue.

Reply:
Sorry I was looking at your comment wrong :pensive:. The condition in the template was `{{#if (not (is-empty this.authMethodOptions))}}`. Looking at it again I suppose that could have been changed `{{#if this.selectedAuthMethod}}` which would have been cleaner but I wasn't sure how all this state was being managed. It seemed like updating `this.authMethodOptions` so it reflected the current state was the most straightforward.

Reply:
I agree, this will be the most straightforward particularly since some namespaces won't have auth mounts to filter on, which is why it's that way  in the template. Thanks for the explanation!